node/src/main/scala/org/bitcoins/node/NeutrinoNode.scala

```diff
@@ -39,7 +39,7 @@ case class NeutrinoNode(
       _ <- peerMsgSender.sendGetCompactFilterCheckPointMessage(
         stopHash = bestHash.flip)
     } yield {
-      node
+      node.asInstanceOf[NeutrinoNode]
```

Comment:
can we just use `this`

Reply:
You've got to be careful with things like `this`, if there ever is internal state modified or set by the  `super.start()` method, but you don't return the thing that is returned by `.start()` the state will not returned to the caller of this method. This is how caching bugs are introduced.

In general with programming, your method should return something (`NeutrinoNode` in this case). Even if `this` is the same type as the thing returned from another method, you should always return the thing given to you by the other method as you don't know if it modifies internal state currently, or will modify internal state in the future.

We should probably figure out how to get rid of the `.asInstanceOf[]` though as that is a sign of an anti pattern.

In short, in this instance you are right there isn't a big difference as of now, but we should really strive to return things that are given to us by methods -- like the `Future[Node]` in this case.